test/e2e/tests/test_workspace.py

```diff
@@ -0,0 +1,167 @@
+        assert workspace_resource is not None
+        assert 'status' in workspace_resource
+        assert 'status' in workspace_resource['status']
+        assert 'statusCode' in workspace_resource['status']['status']
+        assert workspace_resource['status']['status']['statusCode'] == 'ACTIVE'
+        condition.assert_synced(workspace_ref)
+
+        # Next, we verify that the AMP server-side workspace values are the same as
+        # defined in the CR. Afterwards, we modify the spec and verify that the AMP 
+        # server-side resource shows the new value of the field. 
+        latest = self.get_workspace(prometheusservice_client, workspace_resource['status']['workspaceID'])
+        assert latest is not None
+        assert latest['workspace']['alias'] == resource_name
+        assert latest['workspace']['tags']['k1'] == 'v1'
+        assert latest['workspace']['tags']['k2'] == 'v2'
+        assert 'k3' not in latest['workspace']['tags']
+    
+        new_alias = resource_name + "_updated"
+        
+        tag_update = {
+            "k1": "v1_updated",
+            "k2": None,
+            "k3": "v3",
+        }
+        
+        expected_tags = {
+            "k1": "v1_updated",
+            "k3": "v3",
+        }
+
+        updates = {
+            "spec": {"alias": new_alias, "tags":  tag_update},
+        }
+
+        k8s.patch_custom_resource(workspace_ref, updates)
+        time.sleep(MODIFY_WAIT_AFTER_SECONDS)
+
+        # wait for the resource to get synced after the patch
+        assert k8s.wait_on_condition(workspace_ref, "ACK.ResourceSynced", "True", wait_periods=MAX_WAIT_FOR_SYNCED_MINUTES)
+        latest = self.get_workspace(prometheusservice_client, workspace_resource['status']['workspaceID'])
+        assert latest is not None
+        assert latest['workspace']['alias'] == new_alias
+
+        tags.assert_equal_without_ack_tags(latest['workspace']['tags'],expected_tags)
+
+
+        # Next we update the tags again, but this time we try to remove all tags
+        tag_update = {
+            "k1": None,
+            "k3": None,
+        }
+        
+        expected_tags = {}
+
+        updates = {
+            "spec": {"tags":  tag_update},
+        }
+        k8s.patch_custom_resource(workspace_ref, updates)
+        time.sleep(MODIFY_WAIT_AFTER_SECONDS)
+
+        # wait for the resource to get synced after the patch
+        assert k8s.wait_on_condition(workspace_ref, "ACK.ResourceSynced", "True", wait_periods=MAX_WAIT_FOR_SYNCED_MINUTES)
+
+        # After resource is synced again, assert that patches are reflected in the AWS resource
+        latest = self.get_workspace(prometheusservice_client, workspace_resource['status']['workspaceID'])
+        assert latest is not None
+        tags.assert_equal_without_ack_tags((latest['workspace']['tags']), expected_tags)
+
+
+        _, deleted = k8s.delete_custom_resource(workspace_ref)
+        assert deleted
```

Comment:
Can you also call `self.get_workspace` here, and assert that here resource got deleted?